index.html

```diff
@@ -1209,6 +1209,148 @@ <h3>
           conventions or limitations of the host operating system.
         </p>
       </section>
+      <section>
+        <h3>
+          `file_handlers` member
+        </h3>
+        <p>
+          The [=manifest's=] <code><dfn data-export="" data-dfn-for=
+          "manifest">`file_handlers`</dfn></code> member is a [=list=] that
+          provides instructions for how the app handles file-opening actions
+          that originate outside of the app itself.
+        </p>
+        <p>
+          The management, presentation and selection of registered
+          file-handling applications is at the discretion of the operating
+          system.
+        </p>
+        <p>
+          To <dfn>process the `file_handlers` member</dfn>, given [=ordered
+          map=] |json:ordered map|, [=ordered map=] |manifest:ordered map|,
+          [=URL=] |manifest_url:URL|:
+        </p>
+        <ol class="algorithm">
+          <li>Let |processedFileHandlers:list| be a new [=list=].
+          </li>
+          <li>Set |manifest|["file_handlers"] to |processedFileHandlers|.
+          </li>
+          <li>If |json|["file_handlers"] doesn't [=map/exist=] or
+          |json|["file_handlers"] is not a [=list=], return.
+          </li>
+          <li>[=list/For each=] |entry:ordered map| of |json|["file_handlers"]:
+            <ol>
+              <li>Let |file_handler:ordered map| be [=process a file handler
+              item=] with |entry|, [=manifest/start_url=], [=manifest/scope=],
+              and |manifest_url|.
+              </li>
+              <li>If |file_handler| is failure, continue.
+              </li>
+              <li>[=list/Append=] |file_handler| to |processedFileHandlers|.
+              </li>
+            </ol>
+          </li>
+        </ol>
+        <p>
+          On [=installation=], a user agent SHOULD run the process to
+          [=register file handlers=].
+        </p>
+        <section>
+          <h3>
+            Registering file handlers
+          </h3>
+          <p>
+            A user agent SHOULD <dfn>register file handlers</dfn> with the host
+            operating system, consistent with:
+          </p>
+          <ul>
+            <li>The app is exposed in appropriate OS surfaces such as the
+            native file browser and other surfaces that display lists of apps
+            that can handle a given file or file type.
+            </li>
+            <li>The app is a candidate for becoming the system's default
+            handler for associated file types.
+            </li>
+          </ul>
+          <p>
+            A user agent MAY truncate the total set of [=file extensions=] to
+            preserve functionality and prevent abuse.
```

Comment:
I wonder if we should explicitly ban some types (e.g., .sys, .exe, etc.)?

Reply:
actually, it would probably be enough to say that the UA may prevent certain file associations...